sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql

```diff
@@ -60,9 +62,37 @@ categorized_events AS (
     event_category = 'topsites'
     AND event_name = 'click'
     AND mozfun.map.get_key(event_details, "is_sponsored") = "true" AS is_sponsored_topsite_click,
+        -- Pocket
+    event_category = 'pocket'
+    AND event_name = 'click' AS is_pocket_click,
+    event_category = 'pocket'
+    AND event_name = 'click'
+    AND mozfun.map.get_key(event_details, "is_sponsored") = "true" AS is_sponsored_pocket_click,
+    event_category = 'pocket'
+    AND event_name = 'click'
+    AND mozfun.map.get_key(event_details, "is_sponsored") != "true" AS is_organic_pocket_click,
+    event_category = 'pocket'
+    AND event_name = 'impression' AS is_pocket_impression,
+    event_category = 'pocket'
+    AND event_name = 'impression'
+    AND mozfun.map.get_key(
+      event_details,
+      "is_sponsored"
+    ) = "true" AS is_sponsored_pocket_impression,
+    event_category = 'pocket'
+    AND event_name = 'impression'
+    AND mozfun.map.get_key(event_details, "is_sponsored") != "true" AS is_organic_pocket_impression,
+    event_category = 'pocket'
+    AND event_name = 'save' AS is_pocket_save,
+    event_category = 'pocket'
+    AND event_name = 'save'
```

Comment:
Does this sort of mapping exist anywhere else? I wonder if it would make sense to extract this logic into a UDF and add tests?

Reply:
This is fairly simple boolean comparison I think it's fine. `mozfun.map.get_key` is doing a lot of the heavy lifting already